Mathlib/Topology/LocalAtTarget.lean

```diff
@@ -65,15 +65,15 @@ alias ClosedEmbedding.restrictPreimage := Set.restrictPreimage_closedEmbedding
 
 theorem Set.restrictPreimage_isClosedMap (s : Set β) (H : IsClosedMap f) :
     IsClosedMap (s.restrictPreimage f) := by
-  rintro t ⟨u, hu, e⟩
-  refine' ⟨⟨_, (H _ (IsOpen.isClosed_compl hu)).1, _⟩⟩
-  rw [← (congr_arg HasCompl.compl e).trans (compl_compl t)]
-  simp only [Set.preimage_compl, compl_inj_iff]
-  ext ⟨x, hx⟩
-  suffices (∃ y, y ∉ u ∧ f y = x) ↔ ∃ y, y ∉ u ∧ f y ∈ s ∧ f y = x by
-    simpa [Set.restrictPreimage, ← Subtype.coe_inj]
-  exact ⟨fun ⟨a, b, c⟩ => ⟨a, b, c.symm ▸ hx, c⟩, fun ⟨a, b, _, c⟩ => ⟨a, b, c⟩⟩
-#align set.restrict_preimage_is_closed_map Set.restrictPreimage_isClosedMap
+  intro t
+  simp [isClosed_induced_iff]
```

Comment:
Non-terminal `simp` should be avoided because they make proofs hard to maintain. If someone adds a simp lemma in a file imported by this file then the effect of the above line can change, and then the error on the next line will be painful to debug. 

There are various ways to fix that. 

The lawyer style is to merge those last two lines into `simpa [isClosed_induced_iff] using fun u hu e ↦ ⟨f '' u, H u hu, by simp [← e, image_restrictPreimage]⟩`. This is technically legal because you don’t have a non-terminal `simp`. But of course it suffers from the exact same maintainability issue.

The really explicit way is to replace the `simp` line with:
```lean
  suffices ∀ u, IsClosed u → Subtype.val ⁻¹' u = t → 
    ∃ v, IsClosed v ∧ Subtype.val ⁻¹' v = restrictPreimage s f '' t by 
      simpa [isClosed_induced_iff]
```
which is making explicit what your simp is doing.

The intermediate way is to force the list of lemmas used by simp (queried using `simp?`). That would replace your `simp` line with `simp only [isClosed_induced_iff, forall_exists_index, and_imp]`. It is not nice to see but it solves the maintainability issue.

Reply:
Thank you for your comment. I chose to make explicit what the simp is doing.